### PR TITLE
(torchx/workspace) Improve docker push logging using bash escape sequence (mimicks docker CLI) to prevent log spamming when pushing for the first time

### PR DESCRIPTION
**Pull request by kiukchung**
See screenshot for before and after

Before:
image

After:
image

image

### CODE REVIEW

**Comment by kiukchung** [comment]
@kurman , @priyaramani , @d4l3k - can one of you take a look?

**Review by kurman — APPROVED** [review]
Nice, looks much better.

File: torchx/workspace/docker_workspace.py [hunk 1]
```diff
@@ -166,13 +167,42 @@ def push_images(self, images_to_push: Dict[str, Tuple[str, str]]) -> None:
             log.info(f"pushing image {repo}:{tag}...")
             img = client.images.get(local)
             img.tag(repo, tag=tag)
-            for line in client.images.push(repo, tag=tag, stream=True, decode=True):
+            ids = []
+            for event in client.images.push(repo, tag=tag, stream=True, decode=True):
```
Comment (d4l3k) [inline torchx/workspace/docker_workspace.py:171]: can we pull this printing logic out into a helper method that takes the iterator from push?

would also be nice to have a unit test

**Review by d4l3k — CHANGES_REQUESTED** [review]
Thanks for contributing! This looks really nice to have just needs a couple of small changes :)

File: torchx/workspace/docker_workspace.py (hunk 1 above)
Comment (kiukchung) [reply torchx/workspace/docker_workspace.py]: Done. Pulled the print logic out to a function called `print_push_event()` and added unittest for it with a sample event file.

**Review by d4l3k — APPROVED** [review]
thanks for the test!

**Pull request closed without merging**